server/pulp/server/managers/repo/unit_association_query.py

```diff
@@ -349,18 +350,26 @@ def _associated_units_by_type_cursor(unit_type_id, criteria, associated_unit_ids
         :type associated_unit_ids: list
         :rtype: pymongo.cursor.Cursor
         """
-
+        model = get_unit_model_by_id(unit_type_id)
         collection = types_db.type_units_collection(unit_type_id)
```

Comment:
> Also, what if there is not mongoengine model for the content type?

I didn't consider that, perhaps foolishly assuming that we have mongoengine models for every content type.

All the hasattr fun is going away with another issue and will just be sane, but given the possibility that we might not have a model to serialize, this all got convoluted enough for me to want to DRY it out some more.